### ISSUE TITLE
Re-boot/Restart device stop calling periodic calls.

### ISSUE THREAD

**Issue by Ramkesh9** [simplymadeapps/QuickPeriodicJobScheduler#3]
when i do restart the device, then it is not Re-Starting the service automatically which i've assigned periodically as earlier.

**Comment by stephenruda** [comment]
@Ramkesh9 This library will be deprecated soon as I no longer use it in any of my projects and will not be maintaining it.
With that said, you could solve this problem in two ways:

-Fork the project and call [setPersisted]
(https://developer.android.com/reference/android/app/job/JobInfo.Builder#setPersisted(boolean)) on the JobInfo.Builder in QuickPeriodicJobScheduler.java.  Keep in mind you will need the RECEIVE_BOOT_COMPLETED permission.
OR
-Implement your own [BOOT_COMPLETED receiver](http://www.learn-android-easily.com/2013/07/bootcompleted-broadcastreceiver-in.html) and reschedule the jobs once rebooted.

**Comment by Ramkesh9** [comment]
Ok thanks for notifying, 
could you able to suggest me any working?
I also tried 2 more similar library's, but they also have same issue.

**Comment by stephenruda** [comment]
@Ramkesh9 I don't know of other libraries that run periodic jobs more frequently than 15 minutes.  I personally migrated all of my work to use Foreground Services instead.  I have to display a constant notification to display that I am running a foreground service but ultimately it works more consistent.
For example, whatever work you are trying to do with the periodic jobs, you just move it into a Foreground Service that is always running and performing your tasks.

**Comment by Ramkesh9** [comment]
Actually I could not got it completely, i haven't much experience with this technology, but I will try to opt your idea .

**Comment by stephenruda** [comment]
Yes, I personally would recommend just going with a foreground service.  I'd recommend finding a good, recent tutorial on foreground services online.

**Comment by Ramkesh9** [comment]
Ok sure, thank you.

**Comment by Ramkesh9** [comment]
I think this macanism is working on foreground not in background, am i right?

**Comment by stephenruda** [comment]
The library should work in both the foreground and background.  If it is not working in the background you should check the settings that I had linked in the other Issue as the OS may be restricting the app from running in the background.
Ultimately, the best thing to use would be a foreground service because it is the most consistent and should work on every device.
